apix_documentation_swagger_v4.yaml

```diff
@@ -0,0 +1,1186 @@
+        kind:
+          type: string
+          description: Challenge's type (question, code, image, image_choice)
+        created_at:
+          type: string
+          format: date-time
+          description: Timestamp of when the challenge was created
+        expires_at:
+          type: string
+          format: date-time
+          description: Delay of expiration; after this timestamp the answer will
+            be rejected. It can be null.
+
+    MFA_Challenge_Question:
+      description: Structure of an MFA Challenge of type 'question'. Secret
+        questions are often asked in addition to an end user's password.
+      allOf:
+        - $ref: '#/components/schemas/GenericData'
+      type: object
+      required:
+        - attributes
+      properties:
+        attributes:
+          type: object
+          allOf:
+            - $ref: '#/components/schemas/MFA_Challenge'
+            - type: object
+              required:
+                - prompt
+              properties:
+                prompt:
+                  type: string
+                  description: The question to show to the owner of the account
+                  example: What is 4 + 4?
+
+    MFA_Challenge_Code:
+      description: Structure of an MFA Challenge of type 'code'. Verification
+        codes are sometimes sent to an end user's mobile phone before they are
+        able to log in. Verification codes cannot be re-used.
+      allOf:
+        - $ref: '#/components/schemas/GenericData'
+      type: object
+      required:
+        - attributes
+      properties:
+        attributes:
+          type: object
+          allOf:
+            - $ref: '#/components/schemas/MFA_Challenge'
+            - type: object
+              required:
+                - prompt
+              properties:
+                prompt:
+                  type: string
+                  description: The instructions to show to the owner of the
+                    account
+                  example: Please provide the code sent to your phone.
+
+    MFA_Challenge_Image:
+      description: Structure of an MFA Challenge of type 'image'
+      allOf:
+        - $ref: '#/components/schemas/GenericData'
+      type: object
+      required:
+        - attributes
+      properties:
+        attributes:
+          type: object
+          allOf:
+            - $ref: '#/components/schemas/MFA_Challenge'
+            - type: object
+              required:
+                - prompt
+              properties:
+                prompt:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      description: "The type of the image: url or uri"
+                    value:
+                      type: string
+                      description: The URL of the image for url or the content
+                        of the image for uri
+                      example: http://via.placeholder.com/350x150
```

Comment:
We no longer have URLS as the image, none of them can have

Reply:
At least in Plunderer::Faker::Client it looks like url is still supported. @jdsampayo can you confirm please?

Reply:
I confirmed, URL and URI working